tests/debugger/test_debugger_pii.py

```diff
@@ -4,191 +4,203 @@
-            # "refreshtoken",
-            # "remoteaddr",
-            # "routingnumber",
-            "salt",
-            "secret",
-            # "secretkey",
-            # "secrettoken",
-            # "securityanswer",
-            # "securitycode",
-            # "securityquestion",
-            # "serviceaccountcredentials",
-            "session",
-            "sessionid",
-            # "sessionkey",
-            # "setcookie",
-            "signature",
-            # "signaturekey",
-            "sshkey",
-            "ssn",
-            "symfony",
-            # "taxidentificationnumber",
-            "telephone",
-            "token",
-            # "transactionid",
-            # "twiliotoken",
-            # "usersession",
-            # "voterid",
-            # "xapikey",
-            # "xauthtoken",
-            # "xcsrftoken",
-            # "xforwardedfor",
-            # "xrealip",
-            "zipcode",
-            # "xsrf",
-            # "xsrftoken",
-        ]
-
-        validate_pii_redaction(redacted_names)
+        base.validate_probes(
+            {"log170aa-acda-4453-9111-1478a6method": "INSTALLED",}
+        )
+
+        base.validate_snapshots(
+            ["log170aa-acda-4453-9111-1478a6method",]
+        )
+
+        self._validate_pii_redaction(redacted_keys)
+
+    def _validate_pii_redaction(self, should_redact_field_names):
+        agent_logs_endpoint_requests = list(interfaces.agent.get_data(path_filters="/api/v2/logs"))
+        not_redacted = []
+        not_found = list(set(should_redact_field_names))
+
+        for request in agent_logs_endpoint_requests:
+            content = request["request"]["content"]
+
+            if content is not None:
+                for content in content:
+                    debugger = content["debugger"]
+
+                    if "snapshot" in debugger:
+                        for field_name in should_redact_field_names:
+                            fields = debugger["snapshot"]["captures"]["return"]["locals"]["pii"]["fields"]
+
+                            if field_name in fields:
+                                not_found.remove(field_name)
+
+                                if "value" in fields[field_name]:
+                                    not_redacted.append(field_name)
+        error_message = ""
+        if not_redacted:
+            not_redacted.sort()
+            error_message = "Fields not properly redacted: " + "".join([f"{item}, " for item in not_redacted])
+
+        if not_found:
+            not_found.sort()
+            error_message += ". Fields not found: " + "".join([f"{item}, " for item in not_found])
+
+        if error_message != "":
+            raise ValueError(error_message)
+
+    def setup_pii_redaction_full(self):
+        self._setup()
+
+    @irrelevant(context.library < "java@1.34", reason="keywords are not fully redacted")
+    @irrelevant(context.library < "dotnet@2.51", reason="keywords are not fully redacted")
```

Comment:
If this aspect wasn't implemented yet, the good decorator is :  

```suggestion
    @missing_feature(context.library < "java@1.34", reason="keywords are not fully redacted")
    @missing_feature(context.library < "dotnet@2.51", reason="keywords are not fully redacted")
```

Reply:
I prefer to leave it as `irrelevant`, since I don't want these tests to run at all if the version does not match. This doesn't occur when tests are marked as `missing_feature`, tests are still running.

Reply:
After our conversation at slack, marked the test as `missing_feature`